Uchu.StandardScripts/AvantGardens/MaelstromVacuum.cs

```diff
@@ -11,30 +12,61 @@ public class MaelstromVacuum : NativeScript
     {
         public override Task LoadAsync()
         {
-            Listen(Zone.OnObject, async obj =>
+            Listen(Zone.OnObject, obj =>
             {
                 if (!(obj is AuthoredGameObject gameObject)) return;
                 
                 if (gameObject.Lot != 14596) return;
+                
+                if (!(gameObject.Author is Player player)) return;
 
-                var samples = Zone.GameObjects.Where(g => g.Lot == 14718).ToList();
+                Delegate onFireServerEventDelegate = null;
+                onFireServerEventDelegate = Listen(player.OnFireServerEvent, (name, message) =>
```

Comment:
The IDE I use suggests doing this, however, it needs to be referenced inside the lambda function as part of the current design so that the event listeners are cleared. Changing it to the recommended method results in a compile error because the lambda function does not see the delegate as a variable because the lambda function is used to create the delegate.